### PR TITLE
[Cache] Document "framework.cache" nodes

### PR DESCRIPTION
**Pull request by geoffrey-brier**

### CODE REVIEW

**Comment by Jean85** [comment]
This needs to be fixed, it appears that the first few changed lines are broken.

Status: needs work

**Comment by geoffrey-brier** [comment]
Sure I can do it, does anybody know why the test fails ? I've been searching for "cache" reference but couldn't find any @Jean85

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -102,6 +102,8 @@ Configuration
     * :ref:`cache <reference-serializer-cache>`
     * :ref:`enable_annotations <reference-serializer-enable_annotations>`
     * :ref:`name_converter <reference-serializer-name_converter>`
+* `cache`_
```
Comment (Jean85) [inline reference/configuration/framework.rst]: Maybe here is why it's failing? It's misaligned...

**Comment by geoffrey-brier** [comment]
I'm currently struggling to know how to write the doc for the cache pools prototype

```rst
* :ref:`cache <reference-cache>`_
    * :ref:`app <reference-cache-app>`_
    * `system`_
    * `directory`_
    * `default_doctrine_provider`_
    * `default_psr6_provider`_
    * `default_redis_provider`_
    * `pools`_
        * `name`_
            * `adapter`_
            * `public`_
            * `default_lifetime`_
            * `provider`_
            * `clearer`_
```

Is this all right ?

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -55,7 +55,7 @@ Configuration
 * `session`_
     * `storage_id`_
     * `handler_id`_
-    * `name`_
+    * :ref:`name <session-name>`
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: What was the reason for this change?

Comment (xabbuh) [inline reference/configuration/framework.rst]: Isn't it enough to only do that for the other added label with the same name?

Comment (geoffrey-brier) [inline reference/configuration/framework.rst]: Tested it and it failed ^^

Comment (xabbuh) [inline reference/configuration/framework.rst]: Too bad, this means that the anchor used currently (http://symfony.com/doc/current/reference/configuration/framework.html#name) won't work in the future. Maybe @wouterj has an idea how we can prevent the loss.

Comment (javiereguiluz) [inline reference/configuration/framework.rst]: @xabbuh adding the `.. _name:` label manually before the existing `name` option wouldn't work?

Comment (xabbuh) [inline reference/configuration/framework.rst]: @javiereguiluz Might cause conflicts as these labels are global and not limited to the current document.

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -1446,6 +1462,120 @@ value.
...
+.. note::
+
+    Your pool name must differ from ``cache.app`` or ``cache.system``.
+
+adapter
+#######
+
+**type**: ``string`` **default**: ``cache.app``
+
+A cache adapter.
```
Comment (javiereguiluz) [inline reference/configuration/framework.rst]: This description should be improved to explain the purpose of this option. Thanks!

**Comment by geoffrey-brier** [comment]
@javiereguiluz Any idea why the deployment to platform.sh fails ? In any case, the PR is good for another review :)

**Comment by javiereguiluz** [comment]
@geoffrey-brier all the builds are failing ... but our friends at Platform.sh have already fixed the issue and provided two possible solutions. We're deciding which one to use.

File: reference/configuration/framework.rst [hunk 4]
```diff
@@ -1446,6 +1462,134 @@ value.
...
+.. tip::
+
+    It might be tough to understand at the beginning, so to avoid confusion remember that all pools perform the
+    same actions but on different medium given the adapter they are based on. Internally, a pool wraps the definition
+    of an adapter.
+
+.. note::
+
+    The framework bundle ships with multiple adapters: apcu, doctrine, system,
+    filesystem, psr6, and redis.
```
Comment (wouterj) [inline reference/configuration/framework.rst]: This note content should be added to the setting description imo

File: reference/configuration/framework.rst [hunk 5]
```diff
@@ -1446,6 +1462,134 @@ value.
...
+.........................
+
+**type**: ``string``
+
+The service name to use as your default Doctrine provider.
+
+default_psr6_provider
+.....................
+
+**type**: ``string`` **default**: ``%kernel.cache_dir%/pools``
```
Comment (wouterj) [inline reference/configuration/framework.rst]: default should be removed

File: reference/configuration/framework.rst [hunk 6]
```diff
@@ -1446,6 +1462,134 @@ value.
...
+
+The path to the cache directory used by services inheriting from the
+``cache.adapter.filesystem`` adapter (including ``cache.app``).
+
+default_doctrine_provider
+.........................
+
+**type**: ``string``
+
+The service name to use as your default Doctrine provider.
```
Comment (wouterj) [inline reference/configuration/framework.rst]: Maybe add the service name of this "default provider": "This is available as ``cache.doctrine`` service."

File: reference/configuration/framework.rst [hunk 7]
```diff
@@ -1446,6 +1462,134 @@ value.
...
+
+default_psr6_provider
+.....................
+
+**type**: ``string`` **default**: ``%kernel.cache_dir%/pools``
+
+The service name to use as your default PSR-6 provider.
```
Comment (wouterj) [inline reference/configuration/framework.rst]: Same here: "This is available as ``cache.psr6`` service"

File: reference/configuration/framework.rst [hunk 8]
```diff
@@ -1446,6 +1462,134 @@ value.
...
+
+default_redis_provider
+......................
+
+**type**: ``string`` **default**: ``redis://localhost``
+
+The dsn to use by the Redis provider.
```
Comment (wouterj) [inline reference/configuration/framework.rst]: ...and here: "This is available as cache.redis service"

File: reference/configuration/framework.rst [hunk 9]
```diff
@@ -1446,6 +1462,134 @@ value.
...
+**type**: ``prototype``
+
+Name of the pool you want to create.
+
+.. note::
+
+    Your pool name must differ from ``cache.app`` or ``cache.system``.
+
+adapter
+#######
```
Comment (wouterj) [inline reference/configuration/framework.rst]: These should be `"` as well

**Review by wouterj — APPROVED** [review]
Looks good! Left a few minor comments that can be fixed during the merge.

**Comment by xabbuh** [comment]
Thank you very much for this great PR @geoffrey-brier and congratulations for your first contribution to the Symfony documentation. I have merged your PR into the `3.2` branch (as Symfony 3.1 is not maintained anymore) and applied @wouterj's comments in af876ebcd2c78101fd52126f9612b9f2c473e02f.

**Comment by geoffrey-brier** [comment]
@xabbuh Thank you, very glad I could help ;)

**Pull request closed without merging**